sequence_task_1/student/include/impl.h

```diff
@@ -20,31 +21,84 @@ class StaticPlaylist
 {
 public:
     /** @todo Member traits */
+    using value_type      = typename Container::value_type;
+    using size_type       = typename Container::size_type;
+    using allocator_type  = typename Container::allocator_type;
+    using difference_type = typename Container::difference_type;
+    using iterator        = typename Container::iterator;
+    using const_iterator  = typename Container::const_iterator;
+    using reference       = typename Container::reference;
+    using const_reference = typename Container::const_reference;
 
     /** @todo Iterators */
+    const_iterator begin()
+    {
+        return m_tracklist.cbegin();
+    }
+
+    const_iterator end()
+    {
+        return m_tracklist.cend();
+    }
 
     /** @todo Constructor from any reversible sequence container */
+    template<class OtherContainer>
+    StaticPlaylist(const OtherContainer& otherConatiner)
+        : m_tracklist(otherConatiner.cbegin(), otherConatiner.cend())
+    {}
 
     /** @todo Assignment from any reversible sequence container */
+    template<class OtherContainer>
+    Container& operator=(const OtherContainer& otherConatiner)
+    {
+        m_tracklist.assign(otherConatiner.cbegin(), otherConatiner.cend());
+        return m_tracklist;
+    }
 
     /** @todo Add track from initializer */
     template<class... Args>
-    const Song& play(Args&&... songData);
+    const Song& play(Args&&... songData)
+    {
+        return *m_tracklist.insert(end(), {Song(std::forward<Args>(songData))...});
```

Comment:
You don't need to construct a temporary object, if you already have the initialization data, that can be used for in-place construction.